release-checklist: minor updates to the release instructions (#1120)
coreos/ignition: .github/ISSUE_TEMPLATE/release-checklist.md

Tagging:
 - [ ] Write release notes in NEWS. Get them reviewed and merged
     - [ ] If doing a branched release, also include a PR to merge the NEWS changes into master

>  - [ ] Ensure your local copy is up to date with master branch (`git@github.com:coreos/ignition.git`)

Reviewer: I think the word `upstream` from https://github.com/coreos/ignition/pull/1120#discussion_r513908112 is useful to include, for clarity.

Tagging:
 - [ ] Write release notes in NEWS. Get them reviewed and merged
     - [ ] If doing a branched release, also include a PR to merge the NEWS changes into master

>  - [ ] Ensure your local copy is up to date with master and should point to the upstream git repository i.e. `git@github.com:coreos/ignition.git`.

Reviewer 2: ```suggestion
 - [ ] Ensure your local copy is up to date with the upstream master branch (`git@github.com:coreos/ignition.git`)
```

 - [ ] Ensure your working directory is clean

>      - [ ] git clean -fd

Reviewer 2: should probably be `git clean -fdx` to ensure that we're in a completely pristine state

Reviewer: Also, this isn't a separate step from the previous line, so let's not ask the operator to check an additional box here.
